### PR TITLE
Split ConfigFragments.scala into multiple files by usage

### PR DESCRIPTION
**Pull request by jerryz123**
The `ConfigFragments.scala` file has grown too large as the dumping ground of assorted integration-level fragments.
This PR splits that file into several `XFragments.scala` files, and moves them to the `config/fragments` subdirectory.

Some of the files are currently quite empty (Ex: `SubsystemFragments.scala`), but I think that is ok.

This also removes firrtl-intepreter and treadle from the commit-on-master check, as those projects are no longer tracked by submodule.

**Type of change**: other enhancement

**Impact**: software change

**Release Notes**
- Split ConfigFragments.scala into several Fragment files

### CODE REVIEW

**Comment by jerryz123** [comment]
Barstools is not on master, as expected

File: generators/chipyard/src/main/scala/config/fragments/TracegenFragments.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package chipyard.config
+
+import freechips.rocketchip.config.{Config, Field, Parameters}
+import tracegen.{TraceGenSystem}
+import chipyard.{BuildSystem}
+import chipyard.clocking.{HasChipyardPRCI}
+
+class TraceGenTop(implicit p: Parameters) extends TraceGenSystem
+  with HasChipyardPRCI
+class WithTracegenSystem extends Config((site, here, up) => {
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/config/fragments/TracegenFragments.scala:10]:
```suggestion
  with HasChipyardPRCI
  
class WithTracegenSystem extends Config((site, here, up) => {
```

File: generators/chipyard/src/main/scala/config/fragments/SubsystemFragments.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import freechips.rocketchip.config.{Config}
+import freechips.rocketchip.subsystem.{BankedL2Key, CoherenceManagerWrapper}
+
+// Replaces the L2 with a broadcast manager for maintaining coherence
+class WithBroadcastManager extends Config((site, here, up) => {
+  case BankedL2Key => up(BankedL2Key, site).copy(coherenceManager = CoherenceManagerWrapper.broadcastManager)
+})
+
+
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/config/fragments/SubsystemFragments.scala:11]:
```suggestion
```

File: generators/chipyard/src/main/scala/config/fragments/RoCCFragments.scala [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  harts: Int*)(gemminiConfig: GemminiArrayConfig[T,U,V] = GemminiConfigs.defaultConfig) extends Config((site, here, up) => {
+  case MultiRoCCKey => up(MultiRoCCKey, site) ++ harts.distinct.map { i =>
+    (i -> Seq((p: Parameters) => {
+      implicit val q = p
+      val gemmini = LazyModule(new Gemmini(gemminiConfig))
+      gemmini
+    }))
+  }
+})
+
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/config/fragments/RoCCFragments.scala:72]:
```suggestion
```

File: generators/chipyard/src/main/scala/config/fragments/RoCCFragments.scala [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        (i -> Seq((p: Parameters) => {
+          val hwacha = LazyModule(new Hwacha()(p))
+          hwacha
+        }))
+      }
+    }
+  })
+)
+
+class WithMultiRoCCGemmini[T <: Data : Arithmetic, U <: Data, V <: Data](
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/config/fragments/RoCCFragments.scala:62]: Can you somehow reference the Hwacha comment saying this is a similar API?

**Review by abejgonzalez — APPROVED** [review]
LGTM.

**Review by abejgonzalez — APPROVED** [review]
LGTM

**Pull request merged**